### PR TITLE
chore: update `module` and `files` properties in the root package

### PR DESCRIPTION
**Pull request by eunjae-lee**
**Summary**

The root package `react-instantsearch` now has the correct value for the `module` property and includes the `files` property.

* related: https://github.com/vitejs/vite/issues/1488

**Result**

This package gets imported without error in vite.

```js
import { createConnector } from 'react-instantsearch';
```

↑ That code triggered the following error:

```
 > node_modules/vite/dist/node/chunks/dep-e9a16784.js:32563:14: error: [plugin: vite:dep-scan] Failed to resolve entry for package "react-instantsearch". The package may have incorrect main/module/exports specified in its package.json.
    32563 │         throw new Error(`Failed to resolve entry for package "${id}". ` +
          ╵               ^
    at resolvePackageEntry (/Users/eunjaelee/workspace/custom-widget-boilerplate-react/node_modules/vite/dist/node/chunks/dep-e9a16784.js:32563:15)
    at tryNodeResolve (/Users/eunjaelee/workspace/custom-widget-boilerplate-react/node_modules/vite/dist/node/chunks/dep-e9a16784.js:32382:11)
    at Context.resolveId (/Users/eunjaelee/workspace/custom-widget-boilerplate-react/node_modules/vite/dist/node/chunks/dep-e9a16784.js:32265:28)
    at Object.resolveId (/Users/eunjaelee/workspace/custom-widget-boilerplate-react/node_modules/vite/dist/node/chunks/dep-e9a16784.js:44211:55)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async resolve (/Users/eunjaelee/workspace/custom-widget-boilerplate-react/node_modules/vite/dist/node/chunks/dep-e9a16784.js:44407:26)
    at async /Users/eunjaelee/workspace/custom-widget-boilerplate-react/node_modules/vite/dist/node/chunks/dep-e9a16784.js:44515:34
    at async callback (/Users/eunjaelee/workspace/custom-widget-boilerplate-react/node_modules/esbuild/lib/main.js:809:28)
    at async handleRequest (/Users/eunjaelee/workspace/custom-widget-boilerplate-react/node_modules/esbuild/lib/main.js:641:30)

   node_modules/vite/dist/node/chunks/dep-e9a16784.js:44505:18: note: This error came from the "onResolve" callback registered here
    44505 │             build.onResolve({
          ╵                   ~~~~~~~~~
    at setup (/Users/eunjaelee/workspace/custom-widget-boilerplate-react/node_modules/vite/dist/node/chunks/dep-e9a16784.js:44505:19)
    at handlePlugins (/Users/eunjaelee/workspace/custom-widget-boilerplate-react/node_modules/esbuild/lib/main.js:733:23)
    at Object.buildOrServe (/Users/eunjaelee/workspace/custom-widget-boilerplate-react/node_modules/esbuild/lib/main.js:1018:7)
    at /Users/eunjaelee/workspace/custom-widget-boilerplate-react/node_modules/esbuild/lib/main.js:1742:17
    at new Promise (<anonymous>)
    at Object.build (/Users/eunjaelee/workspace/custom-widget-boilerplate-react/node_modules/esbuild/lib/main.js:1741:14)
    at Object.build (/Users/eunjaelee/workspace/custom-widget-boilerplate-react/node_modules/esbuild/lib/main.js:1617:51)
    at /Users/eunjaelee/workspace/custom-widget-boilerplate-react/node_modules/vite/dist/node/chunks/dep-e9a16784.js:44368:54
    at Array.map (<anonymous>)

   src/lib/connector.ts:1:32: note: The plugin "vite:dep-scan" was triggered by this import
        1 │ import { createConnector } from 'react-instantsearch';
          ╵                                 ~~~~~~~~~~~~~~~~~~~~~

error when starting dev server:
Error: Build failed with 1 error:
node_modules/vite/dist/node/chunks/dep-e9a16784.js:32563:14: error: [plugin: vite:dep-scan] Failed to resolve entry for package "react-instantsearch". The package may have incorrect main/module/exports specified in its package.json.
    at failureErrorWithLog (/Users/eunjaelee/workspace/custom-widget-boilerplate-react/node_modules/esbuild/lib/main.js:1443:15)
    at /Users/eunjaelee/workspace/custom-widget-boilerplate-react/node_modules/esbuild/lib/main.js:1125:28
    at runOnEndCallbacks (/Users/eunjaelee/workspace/custom-widget-boilerplate-react/node_modules/esbuild/lib/main.js:915:63)
    at buildResponseToResult (/Users/eunjaelee/workspace/custom-widget-boilerplate-react/node_modules/esbuild/lib/main.js:1123:7)
    at /Users/eunjaelee/workspace/custom-widget-boilerplate-react/node_modules/esbuild/lib/main.js:1230:14
    at /Users/eunjaelee/workspace/custom-widget-boilerplate-react/node_modules/esbuild/lib/main.js:606:9
    at handleIncomingPacket (/Users/eunjaelee/workspace/custom-widget-boilerplate-react/node_modules/esbuild/lib/main.js:703:9)
    at Socket.readFromStdout (/Users/eunjaelee/workspace/custom-widget-boilerplate-react/node_modules/esbuild/lib/main.js:573:7)
    at Socket.emit (events.js:314:20)
    at addChunk (_stream_readable.js:298:12)
```

### CODE REVIEW

**Comment by Haroenv** [comment]
This seems correct, but you should use the React InstantSearch DOM package for custom widget boilerplate :)

File: packages/react-instantsearch/package.json [hunk 1]
```diff
@@ -3,7 +3,11 @@
   "version": "6.11.0",
   "description": "⚡ Lightning-fast search for React and React Native apps, by Algolia",
   "main": "index.js",
-  "module": "es/index.js",
+  "module": "dist/es/index.js",
+  "files": [
+    "index.js",
```
Comment (Haroenv) [inline packages/react-instantsearch/package.json:8]:
```suggestion
    "connectors.js",
    "dom.js",
    "index.js",
    "native.js",
    "server.js",
```

**Review by Haroenv — CHANGES_REQUESTED** [review]

**Comment by eunjae-lee** [comment]
I tried the root package because I need `createConnector` but now I've discovered it's exported from the -dom package as well. Then I'll go for it. Thanks! :)

**Comment by Haroenv** [comment]
if it wasn't exported from -dom, you could have used -core next :)

**Review by Haroenv — APPROVED** [review]

**Pull request merged**